api/queue/work/model_service_deployment.go

```diff
@@ -71,32 +73,50 @@ func (depl *ModelServiceDeployment) Deploy(job *queue.Job) error {
 	version := jobArgs.Version
 	project := jobArgs.Project
 	model := jobArgs.Model
+	model.Project = project
+
+	isRedeployment := false
 
 	// Need to reassign destionationURL cause it is ignored when marshalled and unmarshalled
 	if endpoint.Logger != nil {
 		endpoint.Logger.DestinationURL = depl.LoggerDestinationURL
 	}
 
-	model.Project = project
-	log.Infof("creating deployment for model %s version %s with endpoint id: %s", model.Name, endpoint.VersionID, endpoint.ID)
-
-	// copy endpoint to avoid race condition
+	endpoint.RevisionID++
 	endpoint.Status = models.EndpointFailed
+
+	// if inference service name is not empty, it means we are redeploying the endpoint and the endpoint is already running/serving
```

Comment:
Is this condition also true for endpoints that are currently running (the ones without the RevisionID info in the DB)? And what about terminated / failed endpoints? Agree that they should not be considered a “redeployment”. But going by the check here, will the InferenceServiceName in fact be empty for the failed / terminated endpoints ?

Reply:
-> Is this condition also true for endpoints that are currently running (the ones without the RevisionID info in the DB)?

Yes, correct, because the existing endpoints without `revision_id` have `inference_service_name` in the DB.

-> And what about terminated / failed endpoints? Agree that they should not be considered a “redeployment”. But going by the check here, will the InferenceServiceName in fact be empty for the failed / terminated endpoints ?

Existing endpoints (the ones without the `revision_id` in DB) with `terminated` / `failed` status also have `inference_service_name` in the DB. So for the existing endpoint, it will always be marked as redeploy. But for the new endpoint, it will not have `inference_service_name` in DB if the deployment fails, because now we only set inference_service_name if there's at least one successful deployment.